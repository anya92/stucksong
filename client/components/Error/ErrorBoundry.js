import React, { Component } from 'react';
import { element, func, oneOfType } from 'prop-types';
import Loadable from '../HOC/Loadable';

const AsyncError = Loadable({
  loader: () => import('./ErrorComponent'),
});

export default class ErrorBoundry extends Component {
  static propTypes = {
    children: oneOfType([element, func]).isRequired,
  }

  state = {
    hasError: false,
    error: '',
  }

  componentDidCatch(error) {
    this.setState(() => ({ hasError: true, error: error.message }));
  }

  render() {
    if (this.state.hasError) {
      return (
        <AsyncError error={this.state.error} />
      );
    }
    return (
      <div>
        {this.props.children}
      </div>
    );
  }
}
