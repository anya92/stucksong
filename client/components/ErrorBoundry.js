import React, { Component } from 'react';
import { element } from 'prop-types';

import { ErrorContainer } from '../styles/error';

export default class ErrorBoundry extends Component {
  static propTypes = {
    children: element.isRequired,
  }

  state = {
    hasError: false,
    error: '',
  }

  componentDidCatch(error) {
    this.setState(() => ({ hasError: true, error: error.message }));
  }

  render() {
    if (this.state.hasError) {
      return (
        <ErrorContainer>
          <h1>Error <span role="img" aria-label="sad-emoji">ðŸ˜ž</span></h1>
          <h2>Please try again later.</h2>
          <p>Additional info: {this.state.error}</p>
        </ErrorContainer>
      );
    }
    return (
      <div>
        {this.props.children}
      </div>
    );
  }
}
