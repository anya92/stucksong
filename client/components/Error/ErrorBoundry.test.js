import React from 'react';
import { shallow } from 'enzyme';

import ErrorBoundry from './ErrorBoundry';

describe('ErrorBoundry component', () => {
  let wrapper;
  beforeEach(() => {
    const children = <div />;
    wrapper = shallow(<ErrorBoundry children={children} />);
  });

  it('renders correctly if there is no error', () => {
    expect(wrapper.state().hasError).toBe(false);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly if there is an error', () => {
    wrapper.setState({ hasError: true, error: 'Request failed' });
    expect(wrapper).toMatchSnapshot();
  });
});
