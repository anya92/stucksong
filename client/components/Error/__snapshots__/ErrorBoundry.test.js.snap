// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundry component renders correctly if there is an error 1`] = `
<LoadableComponent
  error="Request failed"
/>
`;

exports[`ErrorBoundry component renders correctly if there is no error 1`] = `
<div>
  <div />
</div>
`;
